packages/loot-core/src/platform/server/sqlite/index.electron.ts

```diff
@@ -84,7 +84,16 @@ export async function asyncTransaction(db, fn) {
 }
 
 export function openDatabase(pathOrBuffer) {
-  return new Database(pathOrBuffer);
+  let db = new Database(pathOrBuffer);
+  // Define Unicode-aware LOWER and UPPER implementation.
+  // This is necessary because better-sqlite3 uses SQLite build without ICU support.
+  db.function('UNICODE_LOWER', { deterministic: true }, arg =>
+    arg?.toLowerCase(),
+  );
+  db.function('UNICODE_UPPER', { deterministic: true }, arg =>
+    arg?.toUpperCase(),
+  );
```

Comment:
Can you move this block to a function in a new file so that it can be shared between all the different sqlite implementations?

Reply:
Also can you add a comment linking to the relevant issue that clarifies `deterministic: true` doesn’t currently do anything? (it will be a nontrivial task to bring in any changes that SQL.js ends up making fwiw)

Reply:
> Can you move this block to a function in a new file so that it can be shared between all the different sqlite implementations?

Which block exactly? The code for it is different between `index.electron.ts` and `index.web.ts`.

> Also can you add a comment linking to the relevant issue that clarifies `deterministic: true` doesn’t currently do anything?

It does do something (optimization opportunity for SQLite), it's just that this is the Electron version of the code which isn't really used. It does make sense to mention this somewhere though so I'll add a comment in `index.web.ts` saying that the function should ideally be created with a deterministic flag but SQL.js does not currently support it and I'll link the relevant issue.

Reply:
> Which block exactly? The code for it is different between `index.electron.ts` and `index.web.ts`.

Oops, I hadn’t noticed that! I guess it makes sense to keep them separate then and re-evaluate some sort of more complex logic if we end up with a bunch of functions.


> I'll add a comment in index.web.ts saying that the function should ideally be created with a deterministic flag but SQL.js does not currently support it and I'll link the relevant issue.

👍